python/tests/kernel/test_direct_call_return_kernel.py

```diff
@@ -456,25 +448,34 @@ def simple_tuple_int32_bool_no_args() -> tuple[np.int32, bool]:
 
     with pytest.raises(RuntimeError) as e:
         simple_tuple_int32_bool_no_args()
-    assert 'Use of tuples is not supported in kernels' in str(e.value)
+    # Note: it may make sense to support that if/since we support
+    # the cast for the individual item types.
+    assert 'cannot convert value of type !cc.struct<"tuple" {i64, i1}> to the requested type !cc.struct<"tuple" {i32, i1}>' in str(
+        e.value)
 
     @cudaq.kernel
     def simple_tuple_int32_bool_no_args1() -> tuple[np.int32, bool]:
         return (np.int32(-13), True)
 
-    with pytest.raises(RuntimeError) as e:
-        simple_tuple_int32_bool_no_args1()
-    assert 'Use of tuples is not supported in kernels' in str(e.value)
+    result = simple_tuple_int32_bool_no_args1()
+    # Note: printing the kernel correctly shows the MLIR
+    # values return type as "tuple" {i32, i1}, but we don't
+    # actually create numpy values even when these are requested
+    # in the signature.
```

Comment:
Can we make this an issue if it isn't already?

Reply:
https://github.com/NVIDIA/cuda-quantum/issues/3524